### PR TITLE
chore: Create Prettier configuration

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by mmiszy** [comment]
Related: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26847

File: README.es.md [hunk 1]
```diff
@@ -1,17 +1,16 @@
 # DefinitelyTyped [![Build Status](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped.svg?branch=master)](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped)
 
 [![Join the chat at https://gitter.im/borisyankov/DefinitelyTyped](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/borisyankov/DefinitelyTyped?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg)](https://github.com/prettier/prettier)
```
Comment (PranavSenthilnathan) [inline README.es.md]: Can this be removed? It would be better to reserve badges for content that's more helpful to users.

File: .editorconfig [hunk 2]
```diff
@@ -7,3 +7,10 @@ insert_final_newline = true
 
 [{*.json,*.yml,*.ts,*.tsx,*.md}]
 indent_style = space
+
+[*.md]
+indent_size = 2
+
+# Top-level .prettierrc.json uses tabs
+[/.prettierrc.json]
+indent_style = tab
```
Comment (PranavSenthilnathan) [inline .editorconfig]: It's fine for specific libraries to use tabs if they feel like it, but let's stick with spaces for general DT files.

File: .prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+	"$schema": "http://json.schemastore.org/prettierrc",
+	"trailingComma": "all",
+	"useTabs": true,
```
Comment (PranavSenthilnathan) [inline .prettierrc.json]: As mentioned in the other comment, let's keep it at spaces.

File: .prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+	"$schema": "http://json.schemastore.org/prettierrc",
+	"trailingComma": "all",
```
Comment (PranavSenthilnathan) [inline .prettierrc.json]: It looks like this rule was only applied to a few `.ts` files. Any idea why?

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: .prettierrc.json (hunk 4 above)
Comment (ExE-Boss) [reply .prettierrc.json]: Because I want to spread applying this across several PRs, starting off with my type definitions.

**Comment by damassi** [comment]
@ExE-Boss - Bump

File: .editorconfig (hunk 2 above)
Comment (ExE-Boss) [reply .editorconfig]: But tabs are&nbsp;obviously the&nbsp;superior option, as&nbsp;you are looking at&nbsp;1&nbsp;byte per&nbsp;indentation&nbsp;level (tabs) vs&nbsp;several&nbsp;bytes per&nbsp;indentation&nbsp;level (spaces).

**Comment by orta** [comment]
Alright - this can now be closed as #35672 was merged and added a prettier config to the repo - thanks everyone

**Pull request closed without merging**